contracts/cw721-base/helpers.ts

```diff
@@ -0,0 +1,433 @@
+/*
+ * This is a set of helpers meant for use with @cosmjs/cli
+ * With these you can easily use the cw721 contract without worrying about forming messages and parsing queries.
+ * 
+ * Usage: npx @cosmjs/cli@^0.22 --init https://raw.githubusercontent.com/CosmWasm/cosmwasm-plus/master/contracts/cw20-base/helpers.ts
+ * 
+ * Create a client:
+ *   const client = await useOptions(defaultOptions).setup(password);
+ *   await client.getAccount()
+ * 
+ * Get the mnemonic:
+ *   await useOptions(coralnetOptions).recoverMnemonic(password)
+ * 
+ * If you want to use this code inside an app, you will need several imports from https://github.com/CosmWasm/cosmjs
+ */
+
+const path = require("path");
+
+interface Options {
+  readonly httpUrl: string
+  readonly networkId: string
+  readonly feeToken: string
+  readonly gasPrice: GasPrice
+  readonly bech32prefix: string
+  readonly hdPath: readonly Slip10RawIndex[]
+  readonly faucetToken: string
+  readonly faucetUrl?: string
+  readonly defaultKeyFile: string
+  readonly gasLimits: Partial<GasLimits<CosmWasmFeeTable>>
+}
+ 
+const heldernetOptions: Options = {
+  httpUrl: 'https://lcd.heldernet.cosmwasm.com',
+  networkId: 'hackatom-wasm',
+  feeToken: 'ucosm',
+  gasPrice:  GasPrice.fromString("0.01ucosm"),
+  bech32prefix: 'cosmos',
+  faucetToken: 'UCOSM',
+  faucetUrl: 'https://faucet.heldernet.cosmwasm.com/credit',
+  hdPath: makeCosmoshubPath(0),
+  defaultKeyFile: path.join(process.env.HOME, ".heldernet.key"),
+  gasLimits: {
+    upload: 1500000,
+    init: 600000,
+    register:800000,
+    transfer: 80000,
+  },
+}
+
+const hackatomOptions: Options = {
+  httpUrl: 'https://lcd.heldernet.cosmwasm.com',
+  networkId: 'hackatom-wasm',
+  feeToken: 'ucosm',
+  gasPrice:  GasPrice.fromString("0.025ucosm"),
+  bech32prefix: 'cosmos',
+  faucetToken: 'COSM',
+  faucetUrl: 'https://faucet.cosmwasm.hub.hackatom.dev/credit',
+  hdPath: makeCosmoshubPath(0),
+  defaultKeyFile: path.join(process.env.HOME, ".hackatom.key"),
+  gasLimits: {
+    upload: 1500000,
+    init: 600000,
+    register:800000,
+    transfer: 80000,
+  },
+}
+
+const defaultOptions: Options = {
+  // httpUrl: 'https://lcd.coralnet.cosmwasm.com',
+  httpUrl: "http://localhost:1317",
+  //networkId: process.env.CW_CHAIN_ID,
+  networkId: 'localnet',
+  ///feeToken: 'ushell',
+  feeToken: 'ucosm',
+  gasPrice:  GasPrice.fromString("0.025ucosm"),
+  //bech32prefix: 'coral',
+  bech32prefix: 'cosmos',
+  hdPath: makeCosmoshubPath(0),
+  faucetToken: "SHELL",
+  faucetUrl: "http://localhost",
+  // faucetUrl: process.env.CW_FAUCET,
+  defaultKeyFile: path.join(process.env.HOME, "localnet.key"),
+  gasLimits: {
+    upload: 1500000,
+    init: 600000,
+    register:800000,
+    transfer: 80000,
+  },
+}
+
+interface Network {
+  setup: (password: string, filename?: string) => Promise<SigningCosmWasmClient>
+  recoverMnemonic: (password: string, filename?: string) => Promise<string>
+}
+
+const useOptions = (options: Options): Network => {
```

Comment:
I think with 0.23 this becomes a lot simpler. But good to stick with what is working for now, we (Confio) can upgrade this to 0.23 after it is merged (and you wish to upgrade)